### PR TITLE
Fix upgrade situation 9.1->10.0

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue
refs #27476

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/base.php [hunk 1]
```diff
@@ -583,6 +582,11 @@ public static function init() {
 			self::initSession();
 		}
 		\OC::$server->getEventLogger()->end('init_session');
+
+		// incognito mode for now
+		$uid = \OC::$server->getSession()->get('user_id');
+		\OC::$server->getSession()->set('user_id', null);
```
Comment (PVince81) [inline lib/base.php]: or setIncognitoMode(), I think we had this earlier when running upgrades.

Not 100% sure about the consequences

File: lib/base.php [hunk 2]
```diff
@@ -357,9 +357,8 @@ private static function printUpgradePage() {
 			$tooBig = $apps->isInstalled('user_ldap') || $apps->isInstalled('user_shibboleth');
 			if (!$tooBig) {
 				// count users
-				$stats = \OC::$server->getUserManager()->countUsers();
-				$totalUsers = array_sum($stats);
-				$tooBig = ($totalUsers > 50);
+				$db = new \OC\User\Database();
+				$tooBig = ($db->countUsers() > 50);
```
Comment (PVince81) [inline lib/base.php]: bäh, why is this even in base.php and not in the updater classes

**Comment by PVince81** [comment]
Tested, works.

Having some doubts about the bit where you set the null user

**Comment by PVince81** [comment]
I looked a bit deeper in the code, seems fine 👍

**Comment by davitol** [comment]
:+1: WFM

File: lib/base.php (hunk 2 above)
Comment (DeepDiver1975) [inline lib/base.php]: this is software archeology

**Pull request merged**